### DISCUSSION TITLE
Writing a custom color based plugin

### DISCUSSION THREAD

**Discussion by jacksleight** [tailwindlabs/tailwindcss#6925] [Help] [answered]
I'm writing a custom color based plugin, which I want to work in a similar way to the `fill` plugin:

```js
  fill: ({ matchUtilities, theme }) => {
    matchUtilities(
      {
        fill: (value) => {
          return { fill: toColorValue(value) }
        },
      },
      { values: flattenColorPalette(theme('fill')), type: ['color', 'any'] }
    )
  },
```

However, I can't seem to figure out how to access Tailwind's `toColorValue` and `flattenColorPalette` functions, is there a way to import/require them in `tailwind.config.js`?

**Comment by alexanderbluhm** [comment, marked as answer]
Hi @jacksleight 
You should be able to do this:
```js
const { default: flattenColorPalette } = require('tailwindcss/lib/util/flattenColorPalette')
```
and the same for `toColorValue` (file is `/util/toColorValue`)

**Reply by jacksleight** [reply]
Brilliant , thanks! I was missing the `default:` bit!

**Reply by alamenai** [reply]
I'm getting this Typescript declaration error:

![image](https://github.com/tailwindlabs/tailwindcss/assets/20143684/b228d5d9-b28e-4530-9bb4-332c4bb2c92a)

**Reply by Op-Pautu** [reply]
did u fix it

**Reply by berrnardes** [reply]
Me too

**Reply by maykon-oliveira** [reply]
Mee too

**Reply by adityaa72** [reply]
Create a declaration file name tailwind.d.ts

``` ts
declare module "tailwindcss/lib/util/flattenColorPalette" {
  export default function flattenColorPalette(
    pallette: Record<string, string>,
  ): Record<string, string>;
}

````

**Reply by tremby** [reply]
If a Typescript wizard wants a challenge and can come up with a TS typing which actually figures out the specific return type of `flattenColorPalette` for given inputs, that would be really nice to have for autocompletion!
